fitbenchmarking/parsing/horace_parser.py

```diff
@@ -0,0 +1,194 @@
+                           'variable')
+    elif "HORACE_LOCATION" in os.environ and \
+         "SPINW_LOCATION" not in os.environ:
+
+        raise ParsingError('Could not parse SpinW problem. Please ensure '
+                           'that SPINW_LOCATION is specfied as a environment '
+                           'variable')
+    else:
+        eng.evalc("horace_on")
+
+
+horace_on()
+
+
+class HoraceParser(FitbenchmarkParser):
+    """
+    Parser for a Horace problem definition file.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # A container for improving efficiency in function call if calling
+        # with full input data
+        self._horace_x: typing.Optional[np.ndarray] = None
+        self._horace_w: typing.Optional[str] = None
+        self._horace_msk: typing.Optional[str] = None
+        self._horace_path: typing.Optional[str] = None
+
+    def _get_data_points(self, data_file_path):
+        """
+        Get the data points of the problem from the data file.
+
+        :param data_file_path: The path to the file to load the points from
+        :type data_file_path: str
+
+        :return: data
+        :rtype: dict<str, np.ndarray>
+        """
+        wye_data_f = self._parse_function(self._entries['wye_function'])
+        path = os.path.join(os.path.dirname(self._filename),
+                            wye_data_f[0]['matlab_script'])
+        eng.addpath(os.path.dirname(path))
+        func_name = os.path.basename(path).split('.', 1)[0]
+
+        self._horace_path = os.path.abspath(os.path.dirname(self._filename))
+
+        name = self._entries["name"].replace(' ', '_')
+
+        self._horace_w = f'w_{name}'
+        self._horace_msk = f'msk_{name}'
+
+        eng.evalc(f"[{self._horace_w}, y, e, {self._horace_msk}] ="
+                  f"{func_name}('{data_file_path}','{self._horace_path}')")
+
+        signal = np.array(eng.workspace['y'], dtype=np.float64)
+        error = np.array(eng.workspace['e'], dtype=np.float64)
+
+        add_persistent_matlab_var(self._horace_msk)
+        add_persistent_matlab_var(self._horace_w)
+        eng.evalc(f'global {self._horace_msk}')
+        eng.evalc(f'global {self._horace_w}')
+
+        y = signal.flatten()
+        e = error.flatten()
+        x = np.ones(len(y))
+
+        self._horace_x = x
+        return {'x': x, 'y': y, 'e': e}
+
+    def _create_function(self) -> typing.Callable:
+        """
+        Process the Horace formatted function into a callable.
+
+        Expected function format:
+        function='foreground=filename,p0=...'
+        or
+        function='foreground=filename,p0=... ; background=filename,q0=...'
+
+        :return: A callable function
+        :rtype: callable
+        """
+        # pylint: disable=attribute-defined-outside-init
+        if len(self._parsed_func) == 1:
+            pfg = self._parsed_func[0]
+            p_names = [k for k in pfg if k != 'foreground']
```

Comment:
More for your awareness rather than a fix
```suggestion
            p_names = filter(lambda k: k != 'foreground', pfg)
```
`pfg` isn't a terribly informative name